src/common/guards/resourceGuards/ApiAuthorizationGuard.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  BadRequestException,
+  CanActivate,
+  ExecutionContext,
+  ForbiddenException,
+  Injectable,
+  InternalServerErrorException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Api } from 'src/entities/api.entity';
+import { Repository } from 'typeorm';
+import { ZaLaResponse } from '../../helpers/response';
+
+@Injectable()
+export class ApiAuthorizationGuard implements CanActivate {
+  constructor(
+    @InjectRepository(Api)
+    private readonly apiRepo: Repository<Api>,
+  ) {}
+  async canActivate(context: ExecutionContext): Promise<boolean> {
```

Comment:
There is a guard that does this  `AuthorizationGuard`, extend it if neccessary instead of creating a new one.